fizz_buzz.rb

```diff
@@ -7,23 +7,30 @@ def divisible_by_3?(number)
 def divisible_by_5?(number)
   (number % 5).zero?
 end
+begin
+  print "Enter a number: "
+  num = Integer(gets)
 
-print "Enter a number: "
-num = gets.to_i
-fizzbuzz_array = []
+  if num.negative?
+    puts "You must enter a positive integer"
+  end
 
-num.times do |i|
-  # Because we start counting from 1, add 1
-  current_num = i + 1
+  fizzbuzz_array = []
 
-  entry = ""
-  entry += "Fizz" if divisible_by_3?(current_num)
-  entry += "Buzz" if divisible_by_5?(current_num)
+  num.times do |i|
+    # Because we start counting from 1, add 1
+    current_num = i + 1
 
-  # If the number is divisible neither by 3 nor 5, just use the number
-  entry = current_num.to_s if entry.empty?
+    entry = ""
+    entry += "Fizz" if divisible_by_3?(current_num)
+    entry += "Buzz" if divisible_by_5?(current_num)
 
-  fizzbuzz_array << entry
-end
+    # If the number is divisible neither by 3 nor 5, just use the number
+    entry = current_num.to_s if entry.empty?
+    fizzbuzz_array << entry
+  end
```

Comment:
Your `begin` block is too long. The only line that can throw an exception is `num = Integer(gets)`

Instead, you can do this:
```ruby
begin
  print "Enter a number: "
  num = Integer(gets)
rescue ArgumentError
  # error message here
  retry
end

# Rest of code here
```